sources/packages/backend/libs/services/src/notifications/notification/notification-actions.service.ts

```diff
@@ -261,6 +262,43 @@ export class NotificationActionsService {
     );
   }
 
+  /**
+   * Creates a new ministry notification for each student when a legacy restriction is added to their account.
+   * @param notifications notifications information.
+   * @param auditUserId user that should be considered the one that is causing the changes.
+   * @param entityManager entity manager to execute in transaction.
+   */
+  async saveLegacyRestrictionAddedNotification(
+    notifications: LegacyRestrictionAddedNotification[],
+    auditUserId: number,
+    entityManager?: EntityManager,
+  ): Promise<void> {
+    const templateId = await this.notificationMessageService.getTemplateId(
+      NotificationMessageType.LegacyRestrictionAdded,
+    );
+    const notificationsToSend = notifications.map((notification) => ({
+      userId: notification.userId,
+      messageType: NotificationMessageType.LegacyRestrictionAdded,
+      messagePayload: {
+        email_address: this.gcNotifyService.ministryToAddress(),
+        template_id: templateId,
+        personalisation: {
+          givenNames: notification.firstName ?? "",
+          lastName: notification.lastName,
```

Comment:
This template is replicating an issue from other templates where the given names are completely concatenated with the last name without leaving a space between then, as shown in the PR description. Can we add the space to the template?